api/handlers.go

```diff
@@ -2,32 +2,153 @@ package api
 
 import (
 	"net/http"
+	"strconv"
 
 	"github.com/NebulousLabs/skynet-accounts/database"
+	"github.com/NebulousLabs/skynet-accounts/metafetcher"
+	"gitlab.com/NebulousLabs/errors"
 
 	"github.com/julienschmidt/httprouter"
-	"gitlab.com/NebulousLabs/errors"
 )
 
-// userHandlerGET returns information about an existing user and create it if it
+// userHandler returns information about an existing user and create it if it
 // doesn't exist.
-func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+func (api *API) userHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	WriteJSON(w, u)
+}
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	offset, _ := strconv.Atoi(ps.ByName("offset"))
+	limit, _ := strconv.Atoi(ps.ByName("limit"))
+	ups, err := api.staticDB.UploadsByUser(req.Context(), *u, offset, limit)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// userDownloadsHandler returns all downloads made by the current user.
+func (api *API) userDownloadsHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
```

Comment:
A potentially nicer way to handle this and a good follow-up would be to introduce a `RequireToken` middleware.
Similar to the `RequirePassword` middleware in Sia. 

Then you could extend this method to 
```func (api *API) userDownloadsHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params, token Token) {```

and register it with ```router.GET("/foo/bar", RequireToken(api.userDownloadsHandler))```

This saves you the call to `tokenFromContext` in every handler that needs it.

Reply:
If all we use the token for is fetching users you might also just pass in the User to the handler.

Reply:
Sounds good, added #23 as F/U.